### PR TITLE
logic: bomb walls, and d2 entrance

### LINKED ISSUE #68: [Logic]: entrance rando: bombable walls are one way.
The issue was spotted with Magpie*: there are several bombable doors in LADX, but they only serve as a lock for bombs one-way: the other side is always open, so if you end up on the wrong side, you can pass through. Related area for Martha's Bay is [logic/overworld.py, line 186](https://github.com/daid/LADXR/blob/master/logic/overworld.py#L186), related area for Multi Chest is [line 389](https://github.com/daid/LADXR/blob/master/logic/overworld.py#L389). There are probably other instances of this, but I don't know about them yet.

The issue in the tracker here is that raft return exit should flag raft hut as a check, but it is listed under hell logic instead.

Looking into fixing these ones myself, but I need a shove in the right direction for how to fix it: how would I say "connect back to the main section as a one-way UNLESS you have bombs"?

<details><summary>* rando info & tracker export</summary>

settings: `LhAghsHTWRDKruEIlc`
seed: `E292F3EC1521B28F40CF5A926A2F9A8D`
magpie json export:
```json
{"inventory":{"SHIELD":2,"SWORD":1,"TOADSTOOL":0,"MAGIC_POWDER":0,"MAX_POWDER_UPGRADE":0,"SHOVEL":1,"BOMB":0,"MAX_BOMBS_UPGRADE":0,"BOW":0,"MAX_ARROWS_UPGRADE":0,"FEATHER":1,"ROOSTER":0,"POWER_BRACELET":0,"PEGASUS_BOOTS":1,"FLIPPERS":1,"HOOKSHOT":0,"MAGIC_ROD":0,"BLUE_TUNIC":0,"RED_TUNIC":0,"OCARINA":0,"SONG1":0,"SONG2":0,"SONG3":0,"BOWWOW":0,"BOOMERANG":0,"SEASHELL":3,"TAIL_KEY":0,"ANGLER_KEY":0,"FACE_KEY":0,"BIRD_KEY":0,"SLIME_KEY":0,"GOLD_LEAF":1,"TRADING_ITEM_YOSHI_DOLL":0,"TRADING_ITEM_RIBBON":0,"TRADING_ITEM_DOG_FOOD":0,"TRADING_ITEM_BANANAS":0,"TRADING_ITEM_STICK":0,"TRADING_ITEM_HONEYCOMB":1,"TRADING_ITEM_PINEAPPLE":0,"TRADING_ITEM_HIBISCUS":0,"TRADING_ITEM_LETTER":0,"TRADING_ITEM_BROOM":0,"TRADING_ITEM_FISHING_HOOK":0,"TRADING_ITEM_NECKLACE":1,"TRADING_ITEM_SCALE":0,"TRADING_ITEM_MAGNIFYING_GLASS":0,"INSTRUMENT1":0,"KEY1":2,"NIGHTMARE_KEY1":0,"INSTRUMENT2":0,"KEY2":1,"NIGHTMARE_KEY2":0,"INSTRUMENT3":0,"KEY3":2,"NIGHTMARE_KEY3":0,"INSTRUMENT4":0,"KEY4":1,"NIGHTMARE_KEY4":0,"INSTRUMENT5":0,"KEY5":0,"NIGHTMARE_KEY5":0,"INSTRUMENT6":0,"KEY6":1,"NIGHTMARE_KEY6":0,"INSTRUMENT7":0,"KEY7":0,"NIGHTMARE_KEY7":0,"INSTRUMENT8":0,"KEY8":2,"NIGHTMARE_KEY8":0,"GREAT_FAIRY":0,"KEY9":0,"NIGHTMARE_KEY9":0},"settings":{"py/object":"magpie.LocalSettings","checkSize":"32","mapBrightness":"50","showOutOfLogic":false,"animateChecks":true,"swapMouseButtons":false,"swapItemsAndMap":false,"hideChecked":false,"ignoreHigherLogic":false,"hideVanilla":false,"dungeonItemsTemplate":"compact.html","itemsTemplate":"default.html","customDungeonItems":null,"customItems":null,"showDungeonItemCount":false,"showItemsOnly":false,"highlightItemsOnHover":true,"enableAutotracking":false},"args":{"py/object":"ladxrInterface.getArgs.<locals>.Args","flags":[],"logic":"hard","accessibility":"all","race":false,"heartpiece":true,"seashells":true,"heartcontainers":true,"instruments":false,"tradequest":true,"witch":true,"rooster":true,"boomerang":"gift","dungeon_items":"keysanity","randomstartlocation":true,"dungeonshuffle":true,"entranceshuffle":"insanity","boss":"default","miniboss":"default","goal":"egg","itempool":"","hpmode":"default","hardmode":"none","steal":"default","bowwow":"normal","overworld":"normal","owlstatues":"","superweapons":false,"nagmessages":true,"multiworld":null},"checkedChecks":{"Mabe Village-Tarin's Gift":{"name":"Tarin's Gift","area":"Mabe Village"},"Southern Face Shrine-Under Armos Cave":{"name":"Under Armos Cave","area":"Southern Face Shrine"},"Mabe Village-Bush Field":{"name":"Bush Field","area":"Mabe Village"},"Mabe Village-Well Heart Piece":{"name":"Well Heart Piece","area":"Mabe Village"},"Toronbo Shores-Sword on the Beach":{"name":"Sword on the Beach","area":"Toronbo Shores"},"Kanalet Castle-Darknut, Zol, Bubble Leaf":{"name":"Darknut, Zol, Bubble Leaf","area":"Kanalet Castle"},"Tal Tal Heights-Damp Cave Heart Piece":{"name":"Damp Cave Heart Piece","area":"Tal Tal Heights"},"Mabe Village-Fishing Game Heart Piece":{"name":"Fishing Game Heart Piece","area":"Mabe Village"},"Ukuku Prairie-East of Seashell Mansion Bush":{"name":"East of Seashell Mansion Bush","area":"Ukuku Prairie"},"Kanalet Castle-Bomberman Meets Whack-a-mole Leaf":{"name":"Bomberman Meets Whack-a-mole Leaf","area":"Kanalet Castle"},"Kanalet Castle-In the Moat Heart Piece":{"name":"In the Moat Heart Piece","area":"Kanalet Castle"},"Tail Cave-Four Zol Chest":{"name":"Four Zol Chest","area":"Tail Cave"},"Tail Cave-Hardhat Beetles Key":{"name":"Hardhat Beetles Key","area":"Tail Cave"},"Tail Cave-Pit Button Chest":{"name":"Pit Button Chest","area":"Tail Cave"},"Tail Cave-Two Stalfos, Two Keese Chest":{"name":"Two Stalfos, Two Keese Chest","area":"Tail Cave"},"Tail Cave-Spark, Mini-Moldorm Chest":{"name":"Spark, Mini-Moldorm Chest","area":"Tail Cave"},"Tail Cave-Mini-Moldorm Spawn Chest":{"name":"Mini-Moldorm Spawn Chest","area":"Tail Cave"},"Tail Cave-Feather Chest":{"name":"Feather Chest","area":"Tail Cave"},"Tal Tal Mountains-Access Tunnel Exterior":{"name":"Access Tunnel Exterior","area":"Tal Tal Mountains"},"Turtle Rock-Left of Hinox Zamboni Chest":{"name":"Left of Hinox Zamboni Chest","area":"Turtle Rock"},"Turtle Rock-Vacuum Mouth Chest":{"name":"Vacuum Mouth Chest","area":"Turtle Rock"},"Turtle Rock-Left Vire Key":{"name":"Left Vire Key","area":"Turtle Rock"},"Turtle Rock-Spark, Pit Chest":{"name":"Spark, Pit Chest","area":"Turtle Rock"},"Turtle Rock-Push Block Chest":{"name":"Push Block Chest","area":"Turtle Rock"},"Turtle Rock-Right Lava Chest":{"name":"Right Lava Chest","area":"Turtle Rock"},"Turtle Rock-Zamboni, Two Zol Key":{"name":"Zamboni, Two Zol Key","area":"Turtle Rock"},"Eagle's Tower-Entrance Key":{"name":"Entrance Key","area":"Eagle's Tower"},"Mabe Village-Dog House Dig":{"name":"Dog House Dig","area":"Mabe Village"},"Toronbo Shores-Outside D1 Tree Bonk":{"name":"Outside D1 Tree Bonk","area":"Toronbo Shores"},"Mabe Village-Shop 200 Item":{"name":"Shop 200 Item","area":"Mabe Village"},"Mysterious Woods-Tail Key Chest":{"name":"Tail Key Chest","area":"Mysterious Woods"},"Mysterious Woods-Toadstool":{"name":"Toadstool","area":"Mysterious Woods"},"Tal Tal Mountains-Paphl Cave":{"name":"Paphl Cave","area":"Tal Tal Mountains"},"Tal Tal Mountains-Outside Mad Batter":{"name":"Outside Mad Batter","area":"Tal Tal Mountains"},"Kanalet Castle-Boots Pit":{"name":"Boots Pit","area":"Kanalet Castle"},"Goponga Swamp-Swampy Chest":{"name":"Swampy Chest","area":"Goponga Swamp"},"Koholint Prairie-Heart Piece of Shame":{"name":"Heart Piece of Shame","area":"Koholint Prairie"}},"entranceMap":{"madambowwow":"start_house","kennel":"landfill","ulrira":"banana_seller","mabe_phone":"landfill","start_house":"landfill","shop":"armos_maze_cave","papahl_house_right":"richard_house","papahl_house_left":"obstacle_cave_exit","trendy_shop":"landfill","library":"kennel","banana_seller":"shop","toadstool_entrance":"right_taltal_connector2","prairie_right_cave_top":"right_taltal_connector1","prairie_right_cave_bottom":"castle_main_entrance","obstacle_cave_exit":"castle_upper_left","papahl_entrance":"castle_secret_entrance","multichest_left":"right_taltal_connector6","right_taltal_connector3":"right_taltal_connector5","right_taltal_connector2":"raft_return_enter","right_taltal_connector6":"raft_return_exit","dream_hut":"heartpiece_swim_cave","witch":"landfill","prairie_to_animal_connector":"left_to_right_taltalentrance","seashell_mansion":"landfill","castle_secret_exit":"papahl_house_right","d6_connector_entrance":"papahl_house_left","castle_main_entrance":"multichest_top","castle_secret_entrance":"multichest_left","writes_cave_left":"multichest_right","writes_cave_right":"prairie_madbatter_connector_exit","obstacle_cave_outside_chest":"prairie_madbatter_connector_entrance","writes_house":"landfill","castle_phone":"writes_house","animal_house1":"hookshot_cave","animal_house2":"d8","animal_house3":"prairie_madbatter","animal_house4":"ghost_house","animal_house5":"d1","fire_cave_exit":"obstacle_cave_outside_chest","castle_upper_left":"obstacle_cave_entrance","castle_upper_right":"d7","toadstool_exit":"papahl_exit","left_taltal_entrance":"papahl_entrance","writes_phone":"castle_jump_cave","moblin_cave":"landfill","photo_house":"desert_cave","graveyard_cave_right":"animal_to_prairie_connector","right_taltal_connector5":"prairie_to_animal_connector","right_taltal_connector4":"prairie_right_cave_high","raft_return_exit":"prairie_right_cave_bottom","fire_cave_entrance":"prairie_right_cave_top"},"connections":[{"entrances":["toadstool_entrance","prairie_right_cave_top"],"connector":{"id":"outer_rainbow","name":"Outer Rainbow Cave","entrances":["right_taltal_connector1","right_taltal_connector2"],"obstacleTypes":[],"checks":[]},"label":"A"},{"entrances":["prairie_right_cave_bottom","obstacle_cave_exit"],"connector":{"id":"castle","name":"Kanalet Castle","entrances":["castle_upper_left","castle_main_entrance"],"obstacleTypes":["BOMBS","SWORD"],"checks":["0x2C5","0x2D2","0x2C6"]},"label":"B"},{"entrances":["multichest_left","right_taltal_connector3"],"connector":{"id":"to_d7","name":"Path to D7","entrances":["right_taltal_connector5","right_taltal_connector6"],"obstacleTypes":[],"checks":[]},"label":"C"},{"entrances":["right_taltal_connector2","right_taltal_connector6"],"connector":{"id":"raft_return","name":"Raft Return","entrances":["raft_return_exit","raft_return_enter"],"obstacleTypes":["ONEWAY"],"checks":[]},"label":"D"},{"entrances":["castle_secret_exit","d6_connector_entrance"],"connector":{"id":"quadruplets","name":"Quadruplets' House","entrances":["papahl_house_left","papahl_house_right"],"obstacleTypes":["YOSHI"],"checks":["0x2A6-Trade"]},"label":"E"},{"entrances":["castle_main_entrance","castle_secret_entrance","writes_cave_left"],"connector":{"id":"multichest","name":"To Five Chest Game","entrances":["multichest_left","multichest_right","multichest_top"],"obstacleTypes":["BOMBS"],"checks":[]},"label":"F"},{"entrances":["writes_cave_right","obstacle_cave_outside_chest"],"connector":{"id":"bay_batter","name":"Bay Mad Batter","entrances":["prairie_madbatter_connector_entrance","prairie_madbatter_connector_exit"],"obstacleTypes":["FLIPPERS"],"checks":[]},"label":"G"},{"entrances":["papahl_house_left","fire_cave_exit","castle_upper_left"],"connector":{"id":"mountain_access","name":"Mountain Access","entrances":["obstacle_cave_entrance","obstacle_cave_outside_chest","obstacle_cave_exit"],"obstacleTypes":["SWORD","BOOTS","HOOKSHOT"],"checks":["0x2BB"]},"label":"H"},{"entrances":["toadstool_exit","left_taltal_entrance"],"connector":{"id":"papahl","name":"Path to Papahl","entrances":["papahl_entrance","papahl_exit"],"obstacleTypes":[],"checks":[]},"label":"I"},{"entrances":["graveyard_cave_right","right_taltal_connector5"],"connector":{"id":"under_river","name":"Under the River","entrances":["prairie_to_animal_connector","animal_to_prairie_connector"],"obstacleTypes":["BOOTS"],"checks":[]},"label":"J"},{"entrances":["right_taltal_connector4","raft_return_exit","fire_cave_entrance"],"connector":{"id":"bay_cliff","name":"Martha's Bay Cliff","entrances":["prairie_right_cave_top","prairie_right_cave_bottom","prairie_right_cave_high"],"obstacleTypes":["BOMBS","FEATHER"],"checks":[]},"label":"K"}]}
```
</details>

### PR DESCRIPTION
**Pull request by foxsouns**
fixes #68 (as best as i can tell: checked all the bomb requirements, and only those two were not accounted for). also stumbled upon another wallclip that is helpful in entrance rando: d2 door into swamp.

### CODE REVIEW

File: logic/overworld.py [hunk 1]
```diff
@@ -274,15 +275,15 @@ def __init__(self, options, world_setup, r):
         animal_village.connect(ukuku_prairie, OR(HOOKSHOT, ROOSTER))
         animal_village_connector_left = Location()
         animal_village_connector_right = Location().connect(animal_village_connector_left, PEGASUS_BOOTS)
-        self._addEntrance("prairie_to_animal_connector", ukuku_prairie, animal_village_connector_left, OR(BOMB, BOOMERANG, MAGIC_POWDER, MAGIC_ROD, SWORD)) # passage under river blocked by bush
+        self._addEntrance("prairie_to_animal_connector", ukuku_prairie, animal_village_connector_left, r.bush) # passage under river
```
Comment (daid) [inline logic/overworld.py:278]: This change is wrong, as `r.bush` allows power bracelet, but for some magical reason, you cannot lift bushes that are over stairs.

Comment (foxsouns) [reply logic/overworld.py:278]: oh. thanks, LA, for being inconsistent. mleh.

Comment (foxsouns) [reply logic/overworld.py]: committed the reversion of this.

File: logic/overworld.py [hunk 2]
```diff
@@ -274,15 +275,15 @@ def __init__(self, options, world_setup, r):
         animal_village.connect(ukuku_prairie, OR(HOOKSHOT, ROOSTER))
         animal_village_connector_left = Location()
         animal_village_connector_right = Location().connect(animal_village_connector_left, PEGASUS_BOOTS)
-        self._addEntrance("prairie_to_animal_connector", ukuku_prairie, animal_village_connector_left, OR(BOMB, BOOMERANG, MAGIC_POWDER, MAGIC_ROD, SWORD)) # passage under river blocked by bush
```
Comment (daid) [inline logic/overworld.py:277]: Missing closing bracket?

File: logic/overworld.py [hunk 3]
```diff
@@ -463,8 +465,8 @@ def __init__(self, options, world_setup, r):
             hookshot_cave.connect(hookshot_cave_chest, AND(FEATHER, PEGASUS_BOOTS)) # boots jump the gap to the chest
             graveyard_cave_left.connect(graveyard_cave_right, HOOKSHOT, one_way=True) # hookshot the block behind the stairs while over the pit
             swamp_chest.connect(swamp, None)  # Clip past the flower
-            self._addEntranceRequirement("d2", POWER_BRACELET) # clip the top wall to walk between the goponga flower and the wall
-            self._addEntranceRequirement("d2", COUNT(SWORD, 2)) # use l2 sword spin to kill goponga flowers
```
Comment (daid) [inline logic/overworld.py:467]: Specific reason you removed the sword beam kills?

Comment (foxsouns) [reply logic/overworld.py:467]: Oh. I messed up there. I'll put it back in.

File: logic/overworld.py (hunk 2 above)
Comment (foxsouns) [reply logic/overworld.py:277]: Cool, I missed one. Bound to happen eventually, sigh.

**Comment by foxsouns** [comment]
Oh my god dude, this is a mess on my end. I am going to have to close and re-make a pull request on this. Lovely.

**Pull request closed without merging**